pkg/shares/split_sparse_shares.go

```diff
@@ -22,6 +22,8 @@ func NewSparseShareSplitter() *SparseShareSplitter {
 
 // Write adds the delimited data to the underlying messages shares.
 func (sss *SparseShareSplitter) Write(msg coretypes.Message) {
+	// TODO coretypes.Message likely needs a ShareVersion field so that this
+	// function can contain a switch statement based on it.
```

Comment:
FWIW this motivated https://github.com/celestiaorg/celestia-app/issues/1022

Reply:
I see two options:
1. add `ShareVersion` as a field to `coretypes.Blob`. Pro: easy to do, leaves this code unaffected. Con: leaks share version to celestia-core types
2. do something hacky in celestia-app (similar to [ExtractShareIndexes](https://github.com/rootulp/celestia-app/blob/fe097d9630d2d4f8544b3b043c3362f8c6d5b131/pkg/shares/share_splitting.go#L88)) that extracts the share version associated with a blob from it's MsgPayForBlob.